### ISSUE TITLE
SC2251: suggested fix does not trigger errexit as expected

### ISSUE THREAD

**Issue by bboozzoo** [koalaman/shellcheck#1588]
#### For bugs
- The rule is documented here: https://github.com/koalaman/shellcheck/wiki/SC2251
- Shellcheck installed from snap, `--edge` channel, detailed version:
```
installed:   v0.6.0+git162.36bb1e7            (562) 3MB -
```

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
set -x
set -e
echo foo
! true
{ ! true; }
( ! true )
echo 'still running'
```
The snippet executes like this:
```
+ set -e
+ echo foo
foo
+ true
+ true
+ true
```

#### Here's what shellcheck currently says:
```
In - line 5:
! true
^-- SC2251: This ! is not on a condition and skips errexit. Use { ! ...; } to errexit, or verify usage.

For more information:
  https://www.shellcheck.net/wiki/SC2251 -- This ! is not on a condition and ...
```

Suggested fix to use `{ ! true }` does not seem to work as expected, but `( ! true )` does.

**Comment by koalaman** [comment]
You're absolutely right. `( ! true )` has this effect, but `{ ! true; }` does not. I don't know how that got through. I've updated it to suggest `&& exit 1` instead, and the check has been tightened to only trigger when `set -e` is in effect.

Great catch, thanks!

**Comment by bboozzoo** [comment]
Thank you for the fix!
